### PR TITLE
Dataset ops v1.2

### PR DESCRIPTION
**Pull request by regit**
Update of #5789 addressing comments.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
- Rebase on latest master
- Fix documentation
- Rename `dataset-exist` to `dataset-lookup`

### CODE REVIEW

**Comment by catenacyber** [comment]
@inashivb as this is datasets, should you be reviewing this ?

**Comment by inashivb** [comment]
> @inashivb as this is datasets, should you be reviewing this ?
Sure. I'll test the commands.

File: src/runmode-unix-socket.c [hunk 1]
```diff
@@ -744,6 +744,100 @@ TmEcode UnixSocketDatasetRemove(json_t *cmd, json_t* answer, void *data)
     }
 }
 
+TmEcode UnixSocketDatasetDump(json_t *cmd, json_t *answer, void *data)
+{
+    SCEnter();
+    SCLogDebug("Going to dump datasets");
+    DatasetsSave();
```
Comment (inashivb) [inline src/runmode-unix-socket.c:751]: Should we check if `save` command is used in the loaded datasets? I used it with a `load` command dataset and got the output
```
Success:
"datasets dump done"
```
but it seems to move to next entry if `save` is not found so it did not seem to do anything.

**Review by inashivb — COMMENTED** [review]
Looks great. :) Test well too. One question inline though.

**Comment by suricata-qa** [comment]
Information:
 
 | field | test | baseline | % |
|--|--|--|--| 
| | tlpr1_stats_chk |
| .flow.memuse | 531654848 | 505591168 | 105.16% | 

Pipeline 6126

File: src/runmode-unix-socket.c (hunk 1 above)
Comment (regit) [reply src/runmode-unix-socket.c:751]: I don't think it is an issue as it is just a noop in this case. For example, you can use the command to dump periodically the datasets (via cron) and in this case you don't really control which are the datasets used and don't really want a warning.

**Pull request closed without merging**